locale/merge_csl_locales

```diff
@@ -0,0 +1,180 @@
+        "id-ID",
+        "is-IS",
+        "it-IT",
+        "ja-JP",
+        "km-KH",
+        "ko-KR",
+        "la",
+        "lt-LT",
+        "lv-LV",
+        "mn-MN",
+        "nb-NO",
+        "nl-NL",
+        "nn-NO",
+        "pl-PL",
+        "pt-BR",
+        "pt-PT",
+        "ro-RO",
+        "ru-RU",
+        "sk-SK",
+        "sl-SI",
+        "sr-RS",
+        "sv-SE",
+        "th-TH",
+        "tr-TR",
+        "uk-UA",
+        "vi-VN",
+        "zh-CN",
+        "zh-TW",
+    ]
+
+    number_formats = {}
+
+    for lang in language_packs:
+        url = csl_locale_base.format(ref=LOCALES_GIT_REF, lang=lang)
+        pack_dir = os.path.join(script_dir, "csl_packs", lang)
+
+        if not os.path.exists(pack_dir):
+            os.makedirs(pack_dir)
+
+        pack_file = os.path.join(pack_dir, f"{lang}.xml")
+
+        if use_cache and os.path.isfile(pack_file):
+            print("Using cached file for " + pack_file)
+        else:
+            print("Loading from " + url)
+            with urllib.request.urlopen(url) as response:
+                code = response.getcode()
+                if code != 200:
+                    sys.sterr.write("Got {0} for {1}\n".format(code, url))
+                    return 1
+                with open(pack_file, "wb") as f:
+                    shutil.copyfileobj(response, f)
+
+        xml = ET.parse(pack_file)
+
+        # first, pull out all the translations for "edition", "editions", and "ed."
+        edition_locators = set()
+        for elem in xml.findall(".//{*}term[@name='edition']"):
+            edition_locators.update(get_all_values(elem))
+        edition_locators = list(edition_locators)
+
+        # next, the translations for "-st", "-nd", "-rd", and "-th"
+        short_ordinal_suffixes = set()
+        for term in xml.findall(".//{*}term"):
+            name = term.attrib.get("name", "")
+            value = term.text
+            if not (name.startswith("ordinal") and value):
+                continue
+            short_ordinal_suffixes.add(value)
+            short_ordinal_suffixes.add(strip_superscript_chars(value))
+        short_ordinal_suffixes = list(short_ordinal_suffixes)
+
+        # lastly, the translations for "first", "second", "third", etc.
+        long_ordinals = {}
+        for term in xml.findall(".//{*}term"):
+            name = term.attrib.get("name", "")
+            if not name.startswith("long-ordinal-"):
+                continue
+            long_ordinals[term.text] = int(
+                term.attrib.get("name", "").rsplit("-", 1)[1]
+            )  # parse the "01" in "long-ordinal-01"
+
+        number_formats[lang] = {
+            "locators": {"edition": edition_locators},
+            "ordinals": {"short": short_ordinal_suffixes, "long": long_ordinals},
+        }
+
+    number_formats[
+        "credit"
+    ] = f"Generated from the CSL locales repository <https://github.com/citation-style-language/locales/tree/{LOCALES_GIT_REF}>."
```

Comment:
Let's add `by https://github.com/zotero/zotero-build/blob/master/locale/merge_csl_locales` to clarify that it's generated by this script, for potential future updaters.

No trailing period